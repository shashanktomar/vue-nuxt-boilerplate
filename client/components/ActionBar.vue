<script>
// @flow
// @jsx ActionBar

import Vue from 'vue'
import Component from 'nuxt-class-component'
import { Action } from 'vuex-class'

import type { CommentType } from '~/types'

// @vue/component
@Component()
/**
 * Actions bar component written in jsx.
 * The main idea of this example is to demostrate flexibility of Vue and jsx.
 * We discourage using `jsx` in a real world apps,
 * unless you know what are you doing!
 */
export default class ActionBar extends Vue {
  @Action('fetchComments')
  /**
   * Typed alias to Vuex `fetchComments` action.
   */
  fetchComments: () => Promise<Array<CommentType>>

  /**
   * Render function. It is an equavalent of `<template>` tag.
   *
   * @see https://vuejs.org/v2/guide/render-function.html
   * @returns { VNode } Virtual node to be rendered by Vue.
   */
  render() {
    return (
      <div class={this.$style.actions}>
        <button
          class={this.$style.reload}
          // Try to use `this.fetchComments(12)` instead to raise type errors
          onClick={this.fetchComments}
        >
          reload comments
        </button>
      </div>
    )
  }
}
</script>

<style lang="scss" module>
@import '~/scss/variables';

.actions {
  text-align: center;
}

.reload {
  display: inline-block;
  cursor: pointer;
  outline: none;
  padding: 5px 10px;
  color: $color-white;
  background-color: $button-color-main;
  border-color: $button-color-main;
}
</style>
