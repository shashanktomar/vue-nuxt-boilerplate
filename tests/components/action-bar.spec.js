// @flow

import Vuex from 'vuex'
import { mount, createLocalVue } from '@vue/test-utils'

import ActionBar from '~/components/ActionBar'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('unit tests for ActionBar component', () => {
  let actions
  let store

  beforeEach(() => {
    actions = {
      fetchComments: jest.fn(), // mocking Vuex action call
    }

    store = new Vuex.Store({ actions })
  })

  test('should have one button', () => {
    const wrapper = mount(ActionBar, { store, localVue })

    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.is(ActionBar)).toBeTruthy()
    expect(wrapper.findAll('button')).toHaveLength(1)
  })

  test('should call fetchComments action', () => {
    const wrapper = mount(ActionBar, { store, localVue })

    const input = wrapper.find('button')
    input.trigger('click')

    expect(actions.fetchComments).toHaveBeenCalled()
  })

  test('should match the snapshot', () => {
    const wrapper = mount(ActionBar, { store, localVue })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
