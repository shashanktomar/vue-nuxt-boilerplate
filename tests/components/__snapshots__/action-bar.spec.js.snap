// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit tests for ActionBar component should match the snapshot 1`] = `"<div class=\\"actions\\"><button class=\\"reload\\">reload comments</button></div>"`;
